programs/drift/src/controller/orders.rs

```diff
@@ -1726,13 +1733,19 @@ pub fn fulfill_perp_order_with_match(
 
     let mut total_quote_asset_amount = 0_u64;
     let base_asset_amount_left_to_fill = if amm_wants_to_make && market.amm.amm_jit_is_active() {
-        let jit_base_asset_amount = crate::math::amm_jit::calculate_jit_base_asset_amount(
-            market,
-            base_asset_amount,
-            taker_price,
-            valid_oracle_price,
-            taker_direction,
-        )?;
+        let jit_base_asset_amount = if !taker.orders[taker_order_index].has_limit_price(slot)?
+            && maker_base_asset_amount < taker_base_asset_amount
+        {
+            0
```

Comment:
why would we not want to jit if maker_base_asset_amount < taker_base_asset_amount ?